matrix/buffer.py

```diff
@@ -521,19 +530,20 @@ def _print_message(self, user, message, date, tags, extra_prefix=""):
 
         self.print_date_tags(data, date, tags)
 
-    def message(self, nick, message, date, extra_tags=None, extra_prefix=""):
+    def message(self, user_id, message, date, extra_tags=None, extra_prefix=""):
         # type: (str, str, int, List[str], str) -> None
-        user = self._get_user(nick)
+        user = self._get_user(user_id)
         tags_type = "message_private" if self.type == "private" else "message"
         tags = self._message_tags(user, tags_type) + (extra_tags or [])
         self._print_message(user, message, date, tags, extra_prefix)
 
         user.update_speaking_time(date)
-        self.unmask_smart_filtered_nick(nick)
+        if user.host:
+           self.unmask_smart_filtered_nick(underscore_nonalnum(user.host))
```

Comment:
You don't need to filter the user_id here, the spec disallows white-space, more info [here](https://matrix.org/docs/spec/appendices.html#user-identifiers).  nio should take care of events that contain invalid user_ids.

Reply:
i am more worried about other symbols, like '=' or '@'. Couldn't find in weechat docs if those are allowed so decided not to risk it.

Reply:
I believe they are fine since the IRC plugin tends to put host names into tags. Any UTF-8 string is fine but the tags get separated by a space, so that's why display names are problematic.